File: fix_data_page.py
import os
import re

html_files = [f for f in os.listdir('.') if f.endswith('.html') and os.path.isfile(f)]

for file in html_files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    # Need to find the sidebar links and add data-page attribute
    # They usually look like:
    # <a href="..." class="sidebar-link...">
    #   <span class="sidebar-icon"...>📊</span>
    #   <span>Home</span>
    # </a>

    # Let's replace specifically for each known link
    # For Home: <span>Home</span> or <span data-i18n="nav.home">Home</span>
    
    # Actually, a simpler way is to regex search for the link block containing specific text.
    
    # Home:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?📊.*?</span[\s]*>[\s]*<span[^>]*>Home</span)', r'\1 data-page="home"\2', content, flags=re.DOTALL)
    
    # New Invoice:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?📝.*?</span[\s]*>[\s]*<span[^>]*>New Invoice</span)', r'\1 data-page="new-invoice"\2', content, flags=re.DOTALL)
    
    # History:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?📋.*?</span[\s]*>[\s]*<span[^>]*>History</span)', r'\1 data-page="history"\2', content, flags=re.DOTALL)
    
    # Clients:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?👥.*?</span[\s]*>[\s]*<span[^>]*>Clients</span)', r'\1 data-page="clients"\2', content, flags=re.DOTALL)
    
    # Templates:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?🏢.*?</span[\s]*>[\s]*<span[^>]*>Templates</span)', r'\1 data-page="templates"\2', content, flags=re.DOTALL)
    
    # Pricing:
    content = re.sub(r'(<a[^>]*class="sidebar-link[^>]*?)(>[\s]*<span[^>]*>.*?💎.*?</span[\s]*>[\s]*<span[^>]*>Pricing</span)', r'\1 data-page="pricing"\2', content, flags=re.DOTALL)

    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)

File: fix_data_page.js
const fs = require('fs');
const files = fs.readdirSync('.').filter(f => f.endsWith('.html'));

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;

    // Home
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?📊[^<]*?<\/span>[\s]*<span[^>]*>Home<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="home"' + g2;
    });

    // New Invoice
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?📝[^<]*?<\/span>[\s]*<span[^>]*>New Invoice<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="new-invoice"' + g2;
    });

    // History
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?📋[^<]*?<\/span>[\s]*<span[^>]*>History<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="history"' + g2;
    });

    // Clients
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?👥[^<]*?<\/span>[\s]*<span[^>]*>Clients<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="clients"' + g2;
    });

    // Templates
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?🏢[^<]*?<\/span>[\s]*<span[^>]*>Templates<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="templates"' + g2;
    });

    // Pricing
    content = content.replace(/(<a[^>]*?class="[^"]*?sidebar-link[^"]*?"[^>]*?)(>[\s]*<span[^>]*>[^<]*?💎[^<]*?<\/span>[\s]*<span[^>]*>Pricing<\/span>)/g, (match, g1, g2) => {
        if (g1.includes('data-page=')) return match;
        return g1 + ' data-page="pricing"' + g2;
    });

    if (content !== original) {
        fs.writeFileSync(file, content, 'utf8');
        console.log(`Updated ${file}`);
    }
});
